.github/workflows/funcbench.yml

```
@@ -0,0 +1,56 @@
+on: repository_dispatch
+name: Funcbench Workflow
+jobs:
+  run_funcbench:
+    name: Running funcbench
+    if: github.event.action == 'funcbench_start'
+    runs-on: ubuntu-latest
+    env:
+      AUTH_FILE: ${{ secrets.PROMBENCH_GKE_AUTH }}
+      BRANCH: ${{ github.event.client_payload.BRANCH }}
+      BENCH_FUNC_REGEX: ${{ github.event.client_payload.BENCH_FUNC_REGEX }}
+      CLUSTER_NAME: prombench
```

```suggestion
      CLUSTER_NAME: funcbench
```

I think best to keep those separate.

OTOH would this allow running more than a single funcbench on different PRs? IIRC in `funcbench_nodepool_create` it will create a new node pool for sufficed with the PR number.

---

Yes, it allows funcbench run for different PRs at the same time. We have different namespaces / nodepools for different PRs.
But if we launch `funcbench` for the same PR twice at the same time, it will have an issue. The one which finishes firstly will always interrupt the second one.

---

I think this was somehow solved in prombench so worth checking with @geekodour

---

currently, prombench gh action workflow will just keep showing status pending till the currently running test does not get over. It concludes that the running test is over if all the nodepools in `nodepools.yaml` are deleted.

One could cancel the workflow in the meantime, otherwise, the test will start once the previous test gets completely canceled.

---

> `CLUSTER_NAME: funcbench`

This will mean we'll need to set up another cluster (i.e `funcbench` will have its own `cluster.yml`), what about renaming the current cluster to `test-infra`, and then it'll be intuitive to add different test-infra components to it. Otherwise we can also have funcbench spin up a cluster every time it's run as well but then again will have to take care of subsequent `/funcbench` command in the same PR slightly differently than what `prombench` does currently.

---

if using the same cluster simplifies the setup yes, lets rename it to test-infra.